### ISSUE TITLE
Add ability to update image alt text in 'update' task

### ISSUE THREAD

**Issue by mjordan** [mjordan/islandora_workbench#166]
Currently it's only available in "create" tasks.

**Comment by rosiel** [comment]
I presume it's not available in "add_media" tasks either?

**Comment by mjordan** [comment]
Correct.

**Comment by dmer** [comment]
Hi just wanted to clarify - using this image_alt_text column will allow one to provide alt-text to images coming in via the "file" column only? I had a question from someone wondering if they could include alt-text for thumbnails (coming in as additional media in a "thumbnails" column.)

**Comment by dmer** [comment]
Thanks @mjordan that's helpful information. It seems like one can achieve the desired result w/ the add_media or update_media.

**Comment by mjordan** [comment]
I thought I would start jotting down some use cases for this functionality. Here's one:

> As a repository manage, I want to add the same alt text to all existing image media for a list of node IDs, e.g. Original File, Service File, and Thumbnail file. 

Please add more!

**Comment by dmer** [comment]
As a repo manager, I have a new requirement to put descriptive alt text on all images in the repo available to the public from data in a spreadsheet prepared for me by content managers.

As a repo manager I want to experiment with using image detection to provide descriptive alt-text to images in my repo utilizing automation that generates a CSV with the image ID and descriptive text.

**Comment by mjordan** [comment]
Thanks @dmer. For 

> As a repo manager I want to experiment with using image detection to provide descriptive alt-text to images in my repo utilizing automation that generates a CSV with the image ID and descriptive text.

what role would you see Workbench playing other than using the resulting CSV as input to an `update_media` job (for example)?

**Comment by dmer** [comment]
Yes - that's all I meant - just another way to create CSV input for WB to update with.

**Comment by mjordan** [comment]
The "Alternative text" property (it's not a field as far as I can tell) is only available on "image" media bundles, not "file", etc. Is it safe to assume that the same alt text would always apply to image media regardless of Media Use term applied to them? For example, would someone ever want to app different alt text to "Original file" image media, "Thumbnail image" image media, or "Service file" image media?

**Comment by mjordan** [comment]
Assuming that the same alt text in the input CSV would apply to all "image" media for a given node, I'd like to propose that we add two new tasks, `add_alt_text` and `update_alt_text`, with node IDs and a single alt text value as the two columns in the input CSV. I am proposing to not add this functionality to the `update_media` task since it currently takes media IDs and not node IDs.

Thoughts?

**Comment by mjordan** [comment]
Now that I've caught up on some of the backlogged PRs, I might spend some time this weekend on the approach I sketched out above. Everybody OK with that?

**Comment by dmer** [comment]
That sounds good to me, Mark. Adding or updating alt-text by node ID is what I've needed in the past. I presume that if you update/add alt-text to a Node whose Original File is a TIF, that the alt-text would be added to the service file and thumbnail?

**Comment by mjordan** [comment]
What I am thinking is that for a node ID, Workbench would get all media attached to that node and add the alt text to all of the "image"-type media, so that would catch the service and thumbnail media. Since a TIF is a "file" and not an "image", the original file would not get the alt text.

**Comment by mjordan** [comment]
Did some work last night to implement the core of this approach, and it works as expected. I'll do some more fine tuning over the weekend, add some `--check` logic, and in integration test.

**Comment by mjordan** [comment]
Looks like the maximum length for alt text out of the box in Drupal is 255 characters. Not sure if there are contrib modules, etc. that can extend this but it might be a good idea to provide a config setting to define this when running `--check`.

**Comment by mjordan** [comment]
OK, this is now in the main branch. Docs are at https://mjordan.github.io/islandora_workbench_docs/alt_text/.
